libcloud/httplib_ssl.py

```diff
@@ -238,6 +238,50 @@ def close(self):  # pragma: no cover
         self.response.close()
 
 
+class HttpLibResponseProxy(object):
+    """
+    Provides a proxy pattern around the :class:`requests.Reponse`
+    object to a :class:`httplib.HTTPResponse` object
+    """
+    def __init__(self, response):
+        self._response = response
+
+    def read(self, amt=None):
+        return self._response.text
+
+    def getheader(self, name, default=None):
+        """
+        Get the contents of the header name, or default
+        if there is no matching header.
+        """
+        if name in self._response.headers.keys():
```

Comment:
It's a case insensitive dictionary yes http://docs.python-requests.org/en/master/api/#requests.Response.headers. Although I need to check if that's not just on the protocol method and requesting keys will just ignore it.